proto/buf/alpha/registry/v1alpha1/labels.proto

```diff
@@ -0,0 +1,73 @@
+// Copyright 2020-2023 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package buf.alpha.registry.v1alpha1;
+
+import "google/protobuf/timestamp.proto";
+
+enum LabelNamespace {
+  LABEL_NAMESPACE_UNSPECIFIED = 0;
+  LABEL_NAMESPACE_TAG = 1;
+  LABEL_NAMESPACE_BRANCH = 2;
+  LABEL_NAMESPACE_GIT_COMMIT = 3;
+}
+
+message Label {
+  LabelNamespace namespace = 1;
+  string name = 2;
+}
+
+message LabelValue {
+  string commit_id = 1;
+}
+
+service LabelService {
+  rpc CreateLabel(CreateLabelRequest) returns (CreateLabelResponse);
+  rpc MoveLabel(MoveLabelRequest) returns (MoveLabelResponse);
+  rpc GetLabels(GetLabelsRequest) returns (GetLabelsResponse);
+}
+
+message CreateLabelRequest {
+  Label label = 1;
+  optional string author = 2;
+  optional google.protobuf.Timestamp create_time = 3;
+}
+
+message CreateLabelResponse {
+  LabelValue commit_id = 1;
+}
+
+message MoveLabelRequest {
+  Label label = 1;
+  optional LabelValue from = 2;
+  LabelValue to = 3;
+}
+
+message MoveLabelResponse {}
+
+message GetLabelsRequest {
+  string module_entity_id = 1;
+  optional string name = 2;
+  optional LabelNamespace namespace = 3;
+}
+
+message GetLabelsResponse {
+  message LabelWithValue {
```

Comment:
nit: I generally dislike prepositions in a name. It's suggestive that we're not modeling the problem right.

We started with a definition of Label that contained both a name and value. Now the value is separate but not the name part.

LabelName: name in a namespace
LabelValue: value
Label: a name in a namespace and a value

That would make this message read a bit better:

```proto
message GetLabelsResponse {
  repeated Label labels = 1;
}
```

It'll change other messages but they too remain clear. Like Move could be:

```proto
message MoveLabelRequest {
  LabelName name = 1;
  optional LabelValue from = 2;
  LabelValue to = 3;
}
```